fakenet/listeners/HTTPListener.py

```diff
@@ -370,15 +370,18 @@ def collect_nbi(self, requestline, headers, post_data=None):
         nbi["version"] = version
 
         for line in str(headers).rstrip().split("\n"):
+            # Split atmost 1 time to ensure ':' in attribute values (ex- URLs) are not treated as a new key
             key, value = line.split(":", 1)
```

Comment:
consider using [str.partition](https://docs.python.org/3/library/stdtypes.html#str.partition) when you want to split something once

Reply:
Thanks for the suggestion! Worked upon that :)